### PR TITLE
Support for Right Delegation to Uuid

### PR DESCRIPTION
**Pull request by jonkjetiloye**
## Description
- Added PartiesClient support for 'parties/partylistbyuuid' Register API
- Updated UserProfileLookup model with UserUuid support
- Added Person Uuid, Organization Uuid and Enterprise User Uuid attribute validation
- Updated existing Party test data with uuids

## Related Issue(s)
- #596

## Developer/Reviewer Checklist
- [x] **Your** code builds clean without any errors or warnings
- [x] No changes to config/appsettings or environment variables created in separate linked PR
- [x] Manual testing done (required)
- [x] Relevant integration test added (required for functional changes)
- [ ] Relevant automated test added (required for functional changes)
- [x] All integration tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.AccessManagement.Core/Services/SingleRightsService.cs [hunk 1]
```diff
@@ -536,12 +536,51 @@ private async Task<(DelegationActionResult Result, ServiceResource Resource, Par
...
+                {
+                    toParty = await _contextRetrievalService.GetPartyByUuid(toOrganizationUuid);
+                }
+
+                if (toParty != null && toParty.PartyTypeName != PartyType.Organisation)
+                {
+                    result.Errors.Add("To", $"The provided To attribute value could not be found as a valid organization.");
+                    return (result, resource, null, null);
+                }
+            }
```
Comment (lovoll) [inline src/Altinn.AccessManagement.Core/Services/SingleRightsService.cs:583]: The next if statement: if (toParty == null && toUser == null) 
The error message should be extended by the new posibilities: AltinnXacmlConstants.MatchAttributeIdentifiers.OrganizationUuid, AltinnXacmlConstants.MatchAttributeIdentifiers.EnterpriseUserUuid and
AltinnXacmlConstants.MatchAttributeIdentifiers.PersonUuid

File: test/Altinn.AccessManagement.Tests/Data/UserProfile/00000000-0000-0000-0002-000000010727.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      "InternetAddress": null,
+      "MailingAddress": null,
+      "MailingPostalCode": "",
+      "MailingPostalCity": "",
+      "BusinessAddress": null,
+      "BusinessPostalCode": "",
+      "BusinessPostalCity": "",
+      "UnitStatus": "N"
+    },
+    "PartyId": 50005545,
```
Comment (lovoll) [inline test/Altinn.AccessManagement.Tests/Data/UserProfile/00000000-0000-0000-0002-000000010727.json:32]: Burde PartyUuid vært lagt til på alle Party objektene?

**Review by lovoll — APPROVED** [review]
See comments but overall approved

File: src/Altinn.AccessManagement.Core/Services/ContextRetrievalService.cs [hunk 3]
```diff
@@ -127,6 +127,57 @@ public async Task<List<Party>> GetPartiesAsync(List<int> partyIds, bool includeS
...
+    /// <inheritdoc/>
+    public async Task<Party> GetPartyByUuid(Guid partyUuid, bool includeSubunits = false, CancellationToken cancellationToken = default)
+    {
+        Dictionary<string, Party> parties = await GetPartiesByUuids(partyUuid.SingleToList(), includeSubunits, cancellationToken);
+        parties.TryGetValue(partyUuid.ToString(), out Party result);
+        return result;
+    }
+
+    /// <inheritdoc/>
+    public async Task<Dictionary<string, Party>> GetPartiesByUuids(IEnumerable<Guid> partyUuids, bool includeSubunits = false, CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (jonkjetiloye) [reply src/Altinn.AccessManagement.Core/Services/ContextRetrievalService.cs:139]: Det kunne den vært. Drev å vurderte å skille det ut i en generisk løsning som kunne brukes uavhengig av om key var guid/partyId/etc men kom ikke så langt.

File: test/Altinn.AccessManagement.Tests/Data/UserProfile/00000000-0000-0000-0002-000000010727.json (hunk 2 above)
Comment (jonkjetiloye) [reply test/Altinn.AccessManagement.Tests/Data/UserProfile/00000000-0000-0000-0002-000000010727.json:32]: Ahh, ja. Har vel glemt å oppdatere PartyUuid i de nøstede Party objektene på UserProfile.
Kan oppdatere

**Pull request merged**